lib/extensions/src/lib/extensions.module.ts

```diff
@@ -15,10 +15,15 @@
  * limitations under the License.
  */
 
-import { NgModule } from '@angular/core';
 import { DynamicExtensionComponent } from './components/dynamic-component/dynamic.component';
 import { DynamicTabComponent } from './components/dynamic-tab/dynamic-tab.component';
 import { DynamicColumnComponent } from './components/dynamic-column/dynamic-column.component';
+import { NgModule, ModuleWithProviders, APP_INITIALIZER } from '@angular/core';
+import { AppExtensionService } from './services/app-extension.service';
+
+export function setupExtensions(service: AppExtensionService): Function {
```

Comment:
this will be a bad thing, as it prevents applications controlling the load process. Here's example what will break if you merge this PR: https://github.com/Alfresco/alfresco-content-app/blob/development/src/app/extensions/extension.service.ts#L130

Reply:
Why the application should not control the load process. extension.service.ts what I see in this class of ACA. I guess need to be moved in any single component that exposes this extensibility.
In fact in this moment this service has inside too many and different responsibility.